src/Meziantou.Framework.WPF/Collections/Internals/DispatchedObservableCollection.cs

```diff
@@ -206,54 +232,112 @@ private void EnqueueEvent(PendingEvent<T> @event)
             ProcessPendingEventsOrDispatch();
         }
 
-        private void ProcessPendingEventsOrDispatch()
+        private void EnqueueEvents(IEnumerable<PendingEvent<T>> events)
         {
+            foreach (var @event in events)
+            {
+                _pendingEvents.Enqueue(@event);
+            }
+
+            ProcessPendingEventsOrDispatch();
+        }
+
+        private void ProcessPendingEventsOrDispatch(bool forceProcess = false)
```

Comment:
I think it shoud be `processBatch`. BTW you should assert that `_batchMode` is not null in this case,